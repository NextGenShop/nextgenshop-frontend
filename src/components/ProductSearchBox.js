import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import InputAdornment from "@material-ui/core/InputAdornment";
import SearchIcon from "@material-ui/icons/Search";

const useStyles = makeStyles((theme) => ({
  mb: {
    marginBottom: theme.spacing(2),
  },
}));

export default function ProductSearchBox({ value, onChange }) {
  const classes = useStyles();
  return (
    <TextField
      className={classes.mb}
      id="search"
      label="Search for a product"
      fullWidth
      value={value}
      onChange={onChange}
      InputProps={{
        startAdornment: (
          <InputAdornment position="start">
            <SearchIcon />
          </InputAdornment>
        ),
      }}
    />
  );
}
