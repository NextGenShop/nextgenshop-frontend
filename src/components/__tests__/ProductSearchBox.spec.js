import React from 'react';
import { shallow } from 'enzyme';
import ProductSearchBox from '../ProductSearchBox';

describe('ProductSearchBox', () => {
  test('should correctly render', () => {
    const component = shallow(<ProductSearchBox />);
    expect(component).toMatchSnapshot();
  });
  test('should correctly render with props', () => {
    const component = shallow(
      <ProductSearchBox value='mockValue' onChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
