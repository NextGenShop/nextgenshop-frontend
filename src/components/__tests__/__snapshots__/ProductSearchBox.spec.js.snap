// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSearchBox should correctly render 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "startAdornment": <WithStyles(ForwardRef(InputAdornment))
        position="start"
      >
        <Memo />
      </WithStyles(ForwardRef(InputAdornment))>,
    }
  }
  className="makeStyles-mb-1"
  fullWidth={true}
  id="search"
  label="Search for a product"
/>
`;

exports[`ProductSearchBox should correctly render with props 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "startAdornment": <WithStyles(ForwardRef(InputAdornment))
        position="start"
      >
        <Memo />
      </WithStyles(ForwardRef(InputAdornment))>,
    }
  }
  className="makeStyles-mb-1"
  fullWidth={true}
  id="search"
  label="Search for a product"
  onChange={[Function]}
  value="mockValue"
/>
`;
